### PR TITLE
Initial beans_4_0.xsd

### PR DESCRIPTION
**Pull request by starksm64**
Squash merge this as I could not squash it locally.

### CODE REVIEW

File: api/src/main/resources/beans_4_0.xsd [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+        deployed in bean archives. A library jar, EJB jar,
+        application client jar or rar archive is a bean archive if
+        it has a file named beans.xml in the META-INF directory. The
+        WEB-INF/classes directory of a war is a bean archive if
+        there is a file named beans.xml in the WEB-INF directory of
+        the war. A directory in the JVM classpath is a bean archive
+        if it has a file named beans.xml in the META-INF directory.
+
+        When running in a CDI-lite environment, this is the only configuration value
+        read from a beans.xml file is the beans bean-discovery-mode attribute.
```
Comment (Ladicek) [inline api/src/main/resources/beans_4_0.xsd:79]:
```suggestion
        When running in a CDI Lite environment, the only configuration value
        read from a beans.xml file is the bean-discovery-mode attribute.
```

File: api/src/main/resources/beans_4_0.xsd [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+            <xs:pattern value="\.?[0-9]+(\.[0-9]+)*"/>
+          </xs:restriction>
+        </xs:simpleType>
+      </xs:attribute>
+      <xs:attribute name="bean-discovery-mode" use="optional" default="annotated">
+        <xs:annotation>
+          <xs:documentation>
+            It is strongly recommended you use "annotated". This is now the default and it
+            is also the default as of 4.0 when an empty beans.xml file is seen. When running
+            in a CDI-lite  environment, this is the only aspect of the beans.xml file that
```
Comment (Ladicek) [inline api/src/main/resources/beans_4_0.xsd:108]:
```suggestion
            in a CDI Lite  environment, this is the only aspect of the beans.xml file that
```

**Review by Ladicek — APPROVED** [review]
Looks good, just 2 minor suggestions.

File: api/src/main/resources/beans_4_0.xsd (hunk 2 above)
Comment (starksm64) [reply api/src/main/resources/beans_4_0.xsd]: I have updated CDI-lite to CDI Lite

**Comment by Ladicek** [comment]
I also suggested to fix the "this is the only configuration value read from a beans.xml file is the beans bean-discovery-mode attribute" malformed sentence, but that's a minor thing really.

**Pull request merged**